main/src/com/google/refine/browsing/Engine.java

```diff
@@ -91,7 +91,7 @@ public void setMode(Mode mode) {
         _mode = mode;
     }
 
-    public FilteredRows getAllRows() {
+    public synchronized FilteredRows getAllRows() {
```

Comment:
Sorry for the delay, but I wanted to make sure I wasn't missing something since synchronization is something that is non-trivial to get right.

In this case, I don't think this is going to gain you any advantage.  The *Engine* objects are created per HTTP request using the serialized facet description which is passed in with the request.  Since synchronized methods use the object's default lock, it's only protecting against multiple invocations from within a single HTTP request which would only be as the result of a particularly pathological bug in our code.

I can't ask you to revise the PR since @magdmartin already merged it, so I'll back out those changes by hand.